File: app/src/main/java/com/example/firefighters/adapters/EmergencyPointAdapter.java
package com.example.firefighters.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.firefighters.R;
import com.example.firefighters.models.EmergencyModel;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class EmergencyPointAdapter extends RecyclerView.Adapter<EmergencyPointHolder> {

    Context context;
    ArrayList<EmergencyModel> emergencies;

    public EmergencyPointAdapter(Context context, ArrayList<EmergencyModel> emergencies) {
        this.context = context;
        this.emergencies = emergencies;
    }

    @NonNull
    @NotNull
    @Override
    public EmergencyPointHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.card_point_emergency, parent, false);
        return new EmergencyPointHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull EmergencyPointHolder holder, int position) {
        holder.textName.setText("EM"+emergencies.get(holder.getAdapterPosition()).getId());
    }

    @Override
    public int getItemCount() {
        return emergencies.size();
    }
}

File: app/src/main/java/com/example/firefighters/adapters/EmergencyPointAdapter.kt
package com.example.firefighters.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.firefighters.R
import com.example.firefighters.models.EmergencyModel

class EmergencyPointAdapter(var context: Context, var emergencies: ArrayList<EmergencyModel>) :
    RecyclerView.Adapter<EmergencyPointHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EmergencyPointHolder {
        val view =
            LayoutInflater.from(context).inflate(R.layout.card_point_emergency, parent, false)
        return EmergencyPointHolder(view)
    }

    override fun onBindViewHolder(holder: EmergencyPointHolder, position: Int) {
        holder.textName!!.text = "EM ${emergencies[holder.adapterPosition].id}"
    }

    override fun getItemCount(): Int {
        return emergencies.size
    }
}
